client/state/inline-help/selectors/get-contextual-help-results.js

```diff
@@ -9,4 +11,25 @@ import 'calypso/state/inline-help/init';
  * @param  {object}  state  Global state tree
  * @returns {Array}         List of contextual results based on route
  */
-export default ( state ) => getContextResults( getSectionName( state ) );
+export default ( state ) => {
+	const currentUserId = getCurrentUserId( state );
+	const hasPurchases = hasCancelableUserPurchases( state, currentUserId );
+
+	const sectionName = getSectionName( state );
+	const contextualResults = getContextResults( sectionName );
+
+	const results = contextualResults.filter( ( { post_id } ) => {
+		// Unless searching with Inline Help or on the Purchases section, hide the
+		// "Managing Purchases" documentation link for users who have not made a purchase.
+		if (
+			post_id === 111349 &&
+			! hasPurchases &&
+			! [ 'purchases', 'site-purchases' ].includes( sectionName )
+		) {
+			return false;
+		}
+		return true;
+	} );
```

Comment:
I would do this filtering in the React component, so that `QueryUserPurchases` and `hasCancelableUserPurchases` are paired inside one component.

That's even more performant, as the `getContextualHelpResults` selector is currently not memoized.

Reply:
Agreed 👍🏻

Reply:
addressed in https://github.com/Automattic/wp-calypso/pull/55978/commits/93817a11a0acfce55a803f16fc7883bf4bf72fee